app/models/journeyDomain/item/ItemDomain.scala

```diff
@@ -215,23 +224,40 @@ object ItemDomain {
     AddSupplyChainActorYesNoPage(itemIndex)
       .filterOptionalDependent(identity)(SupplyChainActorsDomain.userAnswersReader(itemIndex))
 
-  def documentsReader(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] =
-    for {
-      transitOperationDeclarationType <- TransitOperationDeclarationTypePage.reader
-      isGBOfficeOfDeparture           <- CustomsOfficeOfDeparturePage.reader.map(_.startsWith(GB))
-      itemDeclarationType             <- DeclarationTypePage(itemIndex).optionalReader
-      isT2OrT2FItemDeclarationType = itemDeclarationType.exists(_.isOneOf(T2, T2F))
-      documents <- DocumentsSection.arrayReader.map(_.validateAsListOf[Document])
-      consignmentLevelPreviousDocumentPresent = documents.exists(
-        x => x.attachToAllItems && x.`type` == Previous
-      )
-      reader <- (transitOperationDeclarationType, isGBOfficeOfDeparture, isT2OrT2FItemDeclarationType, consignmentLevelPreviousDocumentPresent) match {
-        case (T, true, true, true) =>
-          AddDocumentsYesNoPage(itemIndex).filterOptionalDependent(identity)(DocumentsDomain.userAnswersReader(itemIndex))
-        case _ =>
-          DocumentsDomain.userAnswersReader(itemIndex).map(Some(_))
+  private def isConsignmentPreviousDocDefined(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] =
+    DocumentsSection.arrayReader
+      .map {
+        _.validateAsListOf[Document]
+          .exists(
+            x => x.attachToAllItems && x.`type` == Previous
+          )
+      }
+      .flatMap {
+        case true  => AddDocumentsYesNoPage(itemIndex).filterOptionalDependent(identity)(DocumentsDomain.userAnswersReader(itemIndex))
+        case false => DocumentsDomain.userAnswersReader(itemIndex).map(Some(_))
       }
-    } yield reader
+
+  private def addDocumentCheck(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] =
+    ConsignmentAddDocumentsPage.reader.flatMap {
+      case true  => AddDocumentsYesNoPage(itemIndex).filterOptionalDependent(identity)(DocumentsDomain.userAnswersReader(itemIndex))
+      case false => none[DocumentsDomain].pure[UserAnswersReader]
+    }
+
+  private val externalPages: Kleisli[EitherType, UserAnswers, (DeclarationType, Boolean)] = for {
+    consignmentDecType    <- TransitOperationDeclarationTypePage.reader
+    isGBOfficeOfDeparture <- CustomsOfficeOfDeparturePage.reader.map(_.startsWith(GB))
+  } yield (consignmentDecType, isGBOfficeOfDeparture)
+
+  def documentsReader(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] =
```

Comment:
Purely a personal preference but since `addDocumentCheck` and `externalPages` are specific to `documentsReader` I'd like to see those contained within that method